UI Components: Roadmap entry for missing high priority components (#2753)
ILIAS-eLearning/ILIAS: src/UI/ROADMAP.md

## Short Term

### All UI-Elements Step 1

The UI-Framework attempts to be the source for all visual elements in ILIAS and
thus supersede the current templating. The challenge is two-fold: on the one hand
the required elements need to be implemented in the UI-framework, on the other
hand the Components need to use the UI-framework for their actual rendering. 

Not all Components have the same priority. Highest on our list are Components that 
are needed on every screen to render ILIAS with an empty content section (such as Tabs). 
Next are Components, that are needed in many use cases and therefore have a very high
grade of re-use (such as the Toolbar). Due to major work done for ILIAS 6, such as the Standard Layout 

> Component, Meta Bar and Main Bar Why are now able to promote further Components to our list of short 

Reviewer: ```suggestion
Component, Meta Bar and Main Bar we are now able to promote further Components to our list of short 
```

recommend starting with a Workshop with the UI Components coordinators to align various visions
and concepts of the given Component. The following items have the highest priority on 
our list of short term tasks.

#### Outer Content (advanced, variable)
This Component is basically what could be hooked into the [Standard Layout](Component/Layout/Page/Factory.php) as
content (currently provided as array of Legacy Components). Most Probably it should be able to hold the title section 
(not yet part of the UI Components, see below), the Tabs (not yet Part of the UI Components, see below) and the
Inner Content holding the workspace for the current context (not yet Part of the UI Components, see below).

Note; One important aspect here, will be to clarify at some point the relation to the [Global Screen](../GlobalScreen). 

#### Title Section (advanced, variable)
This Component will probably hold the Icon, title, description and the actions (maybe along with the used glyphs) of the 
current context. Note that a major part of the work for this components will be to setup a comprehensive set of rules on 

> when to provide and Icon, restrictions of the Title (lengths, nouns vs verbs etc.), restrictions of the description 

Reviewer: ```suggestion
when to provide an Icon, restrictions of the Title (lengths, nouns vs verbs etc.), restrictions of the description 
```

(lengths, when to use etc.) and nature, amount of the actions etc. Note that there is pre-existing work on those 
subjects: [Feature Wiki](https://docu.ilias.de/goto_docu_wiki_wpage_6080_1357.html). 

> However, this is not decided upon and most certainly up for discussion.

Reviewer: ```suggestion
However, this has not been decided yet and is thus most certainly up for discussion.
```

Note; One important aspect here, will be to clarify at some point the relation to the [Global Screen](../GlobalScreen). 

#### Tabs and Sub Tabs (advanced, variable)
Note that a major part of the work for this Components will be to setup a comprehensive set of rules on the naming of 
Tabs and Sub Tabs (noun vs verbs, length, amount of words etc.) and rules for the usage of Tabs vs Sub Tabs vs Sections
in Forms shown in Tabs. Also, one would have to look into the issue that currently "<-- Back" actions are mixed into 

> the Tabs. We will need to decide, whether we will still use this concept in future.

Reviewer: ```suggestion
the Tabs. We will need to decide, whether we will still use this concept in the future.
```
